### PR TITLE
Simple name changes and additions to annotation-source.yaml

### PR DESCRIPTION
**Pull request by mbrush**
Updates and replaces suggestions in the `mbrush-name_change_suggestions-1` PR. 

- Focuses on simple attribute name changes that afford better VA alignment. 
- Also suggests a couple minor attribute additions. 
- Thornier issues surrounding confidence and evidence level / direction modeling are addressed in separate issue/PR.  

Inline comments provides rationale and summarizes earlier discussion/feedback.

### CODE REVIEW

File: schema/annotation-source.yaml [hunk 1]
```diff
@@ -36,14 +36,14 @@ $defs:
         description: >-
           A quantitative score reflecting the degree of confidence that the information 
           the information entity represents is true.
-      method:
+      specified_by:
```
Comment (ahwagner) [inline schema/annotation-source.yaml:39]: I am fine with this!

File: schema/annotation-source.yaml [hunk 2]
```diff
@@ -36,14 +36,14 @@ $defs:
...
         oneOf:
         - $ref: "#/$defs/Method"
         - $ref_curie: gks.core:CURIE
         description: >-
           A :ref:`Method` that describes all or part of the process through which the information was
           generated.
-      contributions:
+      contribution:
```
Comment (ahwagner) [inline schema/annotation-source.yaml:46]: This is confusing to me. As an array, there are multiple entities here, and `contributions` seems appropriate (similar to `members` in some VRS entities). @mbrush what is the rationale for this change request?

File: schema/annotation-source.yaml [hunk 3]
```diff
@@ -73,12 +73,12 @@ $defs:
         const: Contribution
         default: Contribution
         description: MUST be "Contribution".
-      agent:
+      contributor:
```
Comment (ahwagner) [inline schema/annotation-source.yaml:76]: I will defer to @larrybabb's judgement on this attribute name.

File: schema/annotation-source.yaml [hunk 4]
```diff
@@ -73,12 +73,12 @@ $defs:
...
         $ref: "#/$defs/Agent"
       date:
         type: string
         format: date
-      role:
+      contributor_role:
```
Comment (ahwagner) [inline schema/annotation-source.yaml:81]: I am fine with the use of `contributor_role` over `role` if we are going the contributor role direction, though I think there is an unresolved issue as to whether we should adopt more of an `action` style contribution than a `role` style contribution. In CIViC, records are `created` by `contributor/agent` in a `curator` role, but records are also `revised` by a `contributor/agent` in a `curator` role. It seems off to me that the role changes based on the action taken and would actually prefer to see `role` changed to `action` instead. @mbrush is it possible that we can have _both_ `role` and `action`? That would be a-ok in my book!

File: schema/annotation-source.yaml [hunk 5]
```diff
@@ -96,7 +96,12 @@ $defs:
         description: MUST be "Agent".
       name:
         type: string
-
+     agent_type: 
+        type: string
+        enum: [ "person", "organization", "software agent"]
+        description: >-
+            A specific type of agent the Agent object represents (e.g. a person, organization, 
+            software agent)
```
Comment (ahwagner) [inline schema/annotation-source.yaml:104]: I like this. 👍

File: schema/annotation-source.yaml [hunk 6]
```diff
@@ -147,7 +152,17 @@ $defs:
       An InformationEntity representing an individual piece of data, generated/acquired through methods 
       which reliably produce truthful information about something.
     properties:
-      value:
+      type:
+        type: string
+        const: DataItem
+        default: DataItem
+        description: Must be "DataItem"
+     data_item_type:
```
Comment (mbrush) [inline schema/annotation-source.yaml:160]: It may be useful to be able to specify the specific type of data item the value/unit represents.

File: schema/annotation-source.yaml [hunk 7]
```diff
@@ -147,7 +152,17 @@ $defs:
       An InformationEntity representing an individual piece of data, generated/acquired through methods 
       which reliably produce truthful information about something.
     properties:
-      value:
+      type:
```
Comment (mbrush) [inline schema/annotation-source.yaml:155]: Adding a type field for DataItem (to match type field in all other classes)

File: schema/annotation-source.yaml [hunk 8]
```diff
@@ -96,7 +96,12 @@ $defs:
         description: MUST be "Agent".
       name:
         type: string
-
+     agent_type:
```
Comment (mbrush) [inline schema/annotation-source.yaml:99]: Figured this may be useful to be able to specify if the agent is a person, organization, or software agent.. VA has considered defining specific Agent subtypes for each of these - but is waiting specific uses cases to arise.

File: schema/annotation-source.yaml (hunk 4 above)
Comment (mbrush) [inline schema/annotation-source.yaml:81]: Changed to more descriptive and precise name used in VA model (makes clear that this field holds the role played by the contributor in the Contribution activity - as opposed to the role played by some other entity).

File: schema/annotation-source.yaml (hunk 3 above)
Comment (mbrush) [inline schema/annotation-source.yaml:76]: Changed to name used in VA model.

File: schema/annotation-source.yaml (hunk 2 above)
Comment (mbrush) [inline schema/annotation-source.yaml:46]: Changed to name used in VA model, where convention is for attribute names to be singular (but don't feel strongly).

File: schema/annotation-source.yaml (hunk 1 above)
Comment (mbrush) [inline schema/annotation-source.yaml:39]: Changed to name used in VA IM

File: schema/annotation-source.yaml (hunk 6 above)
Comment (ahwagner) [inline schema/annotation-source.yaml:160]: I am having some difficulty with this pattern. I understand the rationale–we do not want to explicitly enumerate all data types. But I have a strong dislike of two "type" fields in an object, especially since this boils down to a free-text description of the data, and `description` is a property of `InformationEntity` (and is therefore inherited by `DataItem`) that seems to fit this purpose.

**Review by ahwagner — COMMENTED** [review]
@mbrush thank you, this is just what we need to work on this asynchronously ahead of plenary. I have added my thoughts and am interested to hear those of @larrybabb as well.

File: schema/annotation-source.yaml (hunk 7 above)
Comment (ahwagner) [reply schema/annotation-source.yaml:155]: Good catch!

File: schema/annotation-source.yaml (hunk 1 above)
Comment (larrybabb) [reply schema/annotation-source.yaml:39]: Me too

File: schema/annotation-source.yaml (hunk 2 above)
Comment (larrybabb) [reply schema/annotation-source.yaml:46]: It would be helpful to be consistent across vrs, vrsatile and va to be all singular or always plural when multiples are allowed.

@mbrush i think we use plurals in VRS (eg Haplotype.members) so it would be less disruptive to go that way.

File: schema/annotation-source.yaml (hunk 3 above)
Comment (larrybabb) [reply schema/annotation-source.yaml:76]: I'm cool with this.

File: schema/annotation-source.yaml (hunk 4 above)
Comment (larrybabb) [reply schema/annotation-source.yaml:81]: I think we really need the activity type (prefer activity over action). We can add it later. It is computationally complex to infer the activity from the role. I'm ok with contributing roles but it is less useful for my immediate needs.

Comment (larrybabb) [reply schema/annotation-source.yaml:81]: If the contribution is the activity then we should have a 'subtype' or something to specify the precise event that the Contribution represents.

File: schema/annotation-source.yaml (hunk 8 above)
Comment (larrybabb) [reply schema/annotation-source.yaml:99]: Agent_type is fine. It's similar to what we are asking for at the contribution level above (re: activity type). So for consistency maybe we add a contribution_type to support the ability to capture the action, event or activity that the contribution reflects.

I'm good with this consistent approach

File: schema/annotation-source.yaml (hunk 2 above)
Comment (mbrush) [reply schema/annotation-source.yaml:46]: Just how we did things in the RDF/LOD and ontology world - where I was taught to avoid use of plurals in naming properties/attributes.  But I suspect that the convention outside of this context is to use the plural form for attributes taking arrays as values. Totally fine to go with `contributions` here if that is the case.

Comment (mbrush) [reply schema/annotation-source.yaml:46]: Sure - see my reply to Alex about this above.

File: schema/annotation-source.yaml (hunk 8 above)
Comment (larrybabb) [reply schema/annotation-source.yaml:99]: I don't think contribution_type would be an enumeration. Probably a coding

File: schema/annotation-source.yaml (hunk 6 above)
Comment (larrybabb) [reply schema/annotation-source.yaml:160]: I'll defer to you guys. This seems like a path we are heading down with Agent and Contribution (hopefully). But I'm not using this yet so I'm not invested in the decision at this point.

**Review by larrybabb — COMMENTED** [review]
Mostly great

File: schema/annotation-source.yaml (hunk 4 above)
Comment (mbrush) [reply schema/annotation-source.yaml:81]: To clarify, the `contributor_role` in this context is the _role realized by the agent in the act of making their contribution_ (e.g. a submitter role, or editor role, or approver role).  It is specific to their act of contributing to an artifact.  It is _NOT the role/position they hold in their organization_, e.g. a 'curator position'  (but for the record the VA model does have a  dedicated place to capture this role/position should it be important). 

As I've stated, the main reason I am pushing framing this as a role is to be able to utilize the [Contributor Role Ontology (CRO)](https://github.com/data2health/contributor-role-ontology) that other folks in my group have worked on, and the [Contributor Attribution Model (CAM)](https://contributor-attribution-model.readthedocs.io/en/latest/) that uses CRO terms to describe the way agents contribute to research artifacts. With my community coordination and re-use hat on, I would like to align our models here - and we can really re-use CRO or claim CAM alignment if we don't use roles as proposed.

That said, I  understand the reason for confusion here, and the VA model would support your way of doing things if we simply use the existing `Contribution.contribution type` attribute to hold this info.  For example, to describe a submitter contribution, we would say:

````
type: Contribution
contribution_type: submission
contributor: Agent001
date: 2021-07-15
````

If you want can use this pattern for now, and remain VA compliant.  And consider shifting to use of the `contributor_role` attribute in future iterations should we aspire to use CRO and align with the CAM model.

File: schema/annotation-source.yaml (hunk 6 above)
Comment (mbrush) [reply schema/annotation-source.yaml:160]: We define a general pattern in the VA model there classes have an official 'type' (value must be a class from the VA model), and then a second optional attribute to allow specification of a more specific type where useful.  You utilize this pattern a couple times in the model already, e.g.:

````
  type: Agent
  agent_type: [ person, organization, software agent ]

  type: Method
  method_type:  [Variant Interpretation Guideline, Experimental Protocol, . . . ] 
````

If the concern has to do with the 'string' data type - I think we do want this to be an enumeration or Coding, and would be fine to change to whichever you think best here.  Hopefully this would alleviate your reticence.

File: schema/annotation-source.yaml (hunk 4 above)
Comment (ahwagner) [reply schema/annotation-source.yaml:81]: I read up on CRO, and understand the vision of this with respect to CRediT, etc. And I am supportive of the notion of using `contributor_role` to describe the broad types of roles a person may play in the creation of knowledge. However, I agree with @larrybabb that it adds complexity (and potentially reduces precision) to infer the action taken from the role of the contributor, and think it is important–and more correct–to also have contributions capture the contribution activity directly. It sounds like `contribution_type` is acceptable, though–is it okay to express both?

File: schema/annotation-source.yaml (hunk 1 above)
Comment (ahwagner) [reply schema/annotation-source.yaml:39]: addressed in b546010

File: schema/annotation-source.yaml (hunk 8 above)
Comment (mbrush) [reply schema/annotation-source.yaml:99]: @larrybabb I made it an enum because the intent for me is to simply indicate the agent is a person, org, or software.  I gleaned from Alex's explanation that you guys use enums when the permissible values are meant to be a fixed set not to be extended or changed.  If you see the utility of this attribute differently, and would want to put other/more specific terms here, feel free to change it to a Coding.

File: schema/annotation-source.yaml (hunk 4 above)
Comment (mbrush) [reply schema/annotation-source.yaml:81]: OK - sounds like we will use contribution_type instead of contribution_role for now.  As for keeping both, I don't see what this adds, as in my mind these fields capture the exact same information (its just that one frames the value as an activity, and the other as a role realized by the contribution activity).  @ahwagner if you foresee these attributes capturing different types of information, could you provide an example so I can consider if we might keep both?

File: schema/annotation-source.yaml (hunk 6 above)
Comment (ahwagner) [reply schema/annotation-source.yaml:160]: Would we be able to refer to this as a `subtype` instead of `data_item_type`, here and other places where this pattern is used? And yes, a `coding` would be more appropriate for subtypes in my view if this is something other than a `description` to orient the message receiver.

File: schema/annotation-source.yaml (hunk 4 above)
Comment (ahwagner) [reply schema/annotation-source.yaml:81]: I am going to stick with my [original comment](https://github.com/ga4gh/va-spec/pull/86/files#r961156780) on this as the example, which had a 👍 from @larrybabb so I am assuming he sees it the same way. You had mentioned that the role does not capture the organizational role and I get that, but `[curator](https://ontobee.org/ontology/CRO?iri=http://purl.obolibrary.org/obo/CRO_0000027)` is definitely a CRO role and I think the act of accepting / revising / rejecting curated records is still curation, but the corresponding roles for those specific actions (e.g. modifier_role, acceptor_role) are not descendent terms of `curator` so it is unclear that one would know this person was modifying e.g. the _curation_ of knowledge instead of the _generation_ of knowledge. It would be great of `CRO` had the notion of an "expert curator" so we could use CRO to recognize contributions from a person who plays the role of an acknowledged domain expert who is adding / modifying / accepting data in the process of knowledge curation, with the specific actions captured by `contribution_type`. Because at the end of the day (for CIViC, at least), each contribution captured by the system requires review and discussion of the same data, and the role shouldn't be anything other than "curator" just because a person clicks the "accept" button (i.e. agreement) vs. revising / rejecting the initial content vs. providing the initial content.

Comment (mbrush) [reply schema/annotation-source.yaml:81]: My final thoughts here:
 
To start, I am not sure how/why the notion of a **'curator' role/action** came up - as it doesn’t seem to be relevant to our data (we don’t capture this as a contributor role/action in metakb or cvc data AFAIK).  And it is a challenging example on which to base our discussions because it is such a broad and variably used term - and one that is commonly used to describe an organizational job/position title, as well as a type of action that a person may perform (even if they are not officially a 'curator' in their organization). This is why I don't really like using this term to describe an agent's contribution.  Re: its representation in CRO - I do think some work is needed here to address the more specific types of contribution activities/roles that it subsumes. but that is a separate and not pertinent matter.
 
So if it pleases the court, lets completely set the conflated notion of 'curation/curator' aside as not relevant here.  For our purposes, we have already enumerated a defined set of ways that agents can contribute to cvc and metakb statements and other information entities.
 
For the cvc data, the key ones seem to be:
- framed as roles: `submitter`, `evaluator`, `approver`, and `transformer`.
- framed as actions: `submission`, `evaluation`, `approval`, `transformation`
 
For the CIViC data, the key ones seem to be (although I may be missing a couple):
- framed as roles: `submitter`, `reviewer`, `modifier`, `acceptor`
- framed as actions: `submission`, `review`, `modification`, `accepted`
 
We have a VA 'Contribution' pattern that could allow us to frame these _as roles realized in a contribution activity_, e.g.:
````
type: Contribution
contributor: Agent001
contributor_role: approver
date: 2021-07-15
 ````

Or we could chose a VA 'Contribution' pattern that lets us frame these as _a more specific type of contribution activity_, e.g.:

````
type: Contribution
contribution_type: approval
contributor: Agent001
date: 2021-07-15
 ````

While the semantics of these representations are subtly different, they get the same information across.  One pattern lets us use terms from an existing ontology (CRO) to carry this information, and align with the modeling pattern/semantics of the CAM model. The other pattern is still valid in VA - and seems to be more intuitive to metakb and cvc users. 
 
As I said, I am happy to go with either pattern at this point, and revisit when we have additional use cases. What I _don’t think we need_ in any given implementation model is to support both patterns - unless you can provide an example of when this might be needed or useful @ahwagner?

Finally, I'll note that if we did want to represent the organizational position/status that the agent held when they made a particular contribution, the VA  model provides a pattern for this as well - but this doesn't seem to be needed here so I won't get into it.

File: schema/annotation-source.yaml (hunk 6 above)
Comment (mbrush) [reply schema/annotation-source.yaml]: Alex, is your proposal to rename all attributes implementing this pattern using the word 'subtype' instead of 'type'? e.g.:
- `agent_type` --> `agent_subtype`
- `method_type` --> `method_subtype`
- `contribution_type` --> `contribution_subtype`
- `data_item_type` --> `data_item_subtype`
- . . . 
 
Concrete examples of the difference:
 
````
type: Agent
agent_type:  Organization
 vs.
type: Agent
agent_subtype: Organization
````
 
````
type: Method
method_type:  Variant Interpretation Guideline
 vs.
type: Method
method_subtype:  Variant Interpretation Guideline
````
 
````
type: DataItem
data_item_type:  p-value
 vs.
type: DataItem
data_item_subtype:  p-value

Comment (mbrush) [reply schema/annotation-source.yaml]: At the end of the day it's a minor difference and I could go either way, but I would prefer we keep it as is for now and see if it causes any issues with uses.  

For me, I am just not keen on against seeing the 'subtype' in these attribute names.  I've seen/used the general `"type": "X", "x_type": "X1"` pattern and naming approach in many other models and to me it is pretty clear. It didn't seem to raise any hackles on other classes (agent, method, etc) . . . is there something about data item that makes it problematic here? 

Whatever naming pattern we choose, I'm fine to change the value taken by  `data_item_type` to a Coding or enum as Alex suggests.

Comment (larrybabb) [reply schema/annotation-source.yaml]: This is a bit of a minor thing, but the problem is once you make it then you have to live with it, so it may be worth getting it right (with a solid justification).
By going with "subtype" we avoid the confusion with the "type" field.  Contribution.type vs Contribution.contribution_type is confusing on its face. Professionally, I try not to repeat the class name in the attribute names, because every attribute is inherently qualified by the class containing it. 
I can sort of understand how calling an attribute of X an X_<thing> you are really saying that this is a more specific level of the X instance. But has not been how I interpret class names. When I see an attribute with the classname as a prefix, I assume it is an unintentional redundancy.  There have been folks that put the datatype in the attribute names of variables and it has been discouraged as a beginner mistake for the most part (at least in "coding improvement" books and articles I've read over the years). 

I don't mean to blow up any big patterns you've settled on, but some reflection on those attributes that have the class  name in them might be worthwhile to determine if the additional redundant qualifier is meaningful.  Could it be an ontology related need that attribute names are thought of as independent entities and therefore need further specialization? 

In any case, we are trying to devise a schema and a model. and the model should clearly convey that any of the attributes of a component is named as a dependent characteristic of its parent. Thus there should not be any automatic need to repeat the classname in the name of the attribute.

In the end, if we change "type" to "subtype" it is a bit clearer that we are representing the X.subtype of the X.type (and the X.type would always be "X"), so there's a clear difference in the attributes and it should be super clear that subtype is a specialization of type.

Comment (mbrush) [reply schema/annotation-source.yaml]: Thanks for this Larry - I have a clearer sense of your concerns. You guys are closer to the implementation/developer side, so if you prefer a different pattern, I will concede.
 
We could go with 'x_subtype' as Alex suggested.
 
We might also consider 'x_category' . . . 'category' is what FHIR uses when it wants to specify a more specific type (e.g. see  [Substance](https://www.hl7.org/fhir/substance.html), [ResearchStudy](https://www.hl7.org/fhir/researchstudy.html), and [Condition](https://www.hl7.org/fhir/condition.html)).  Not sure I love this however, as I generally consider the notion of a 'category' to be a broader/higher level bucket - and thus not entirely suited for the more granular typing we are trying to do here.

Comment (larrybabb) [reply schema/annotation-source.yaml]: i agree with your concerns over the use of category for this attribute. I think FHIR got it wrong. We should use 'subtype'.  There will be plenty of folks in the future to comment and provide input, we can change as the model matures.
Remember that we will presenting these classes as "Draft" or "Trial Use" which means they are experimental and subject to change.

File: schema/annotation-source.yaml (hunk 8 above)
Comment (larrybabb) [reply schema/annotation-source.yaml]: For contribution_subtype we will likely want a Coding as there may be a bunch of unknown activities that folks will want to identify as contributions.  I do not think we can control this vocabulary out of the gate.
